backend/api/Controllers/ReportController.cs

```diff
@@ -26,9 +26,15 @@ public ReportController(IReportService reportService)
     [ProducesResponseType(StatusCodes.Status401Unauthorized)]
     [ProducesResponseType(StatusCodes.Status403Forbidden)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
-    public async Task<ActionResult<IEnumerable<Report>>> GetReports()
+    public async Task<ActionResult<IList<Report>>> GetReports([FromQuery] string assetCode)
     {
-        var reports = await _reportService.ReadAll();
+        IList<Report> reports;
+        if (assetCode != null)
+            reports = await _reportService.ReadAll(assetCode);
+        else
+            reports = await _reportService.ReadAll();
+        if (reports == null)
+            return NotFound($"Could not find any reports with assetCode {assetCode}");
```

Comment:
Maybe change to more generic message `Could not find any reports matching the query`